PSKoans/Koans/Constructs and Patterns/AboutRegularExpressions.Koans.ps1

```diff
@@ -0,0 +1,468 @@
+            #>
+            [bool]$whitespaceMatch = $____ # Should be either $true or $false
+            $whitespaceMatch | Should -Be 'The moon in June is a big balloon' -match '\s{2,}'
+        }
+    }
+
+    Context '( and ) - Round Braces' {
+        It 'groups patterns together' {
+            <#
+                Round brackets are used to create groups of symbols. These groups can then have
+                quantifiers applied to the entire group, or be used in approximately 42 trillion
+                other ways. This is just the tip of the round bracket/regular expression groups
+                iceberg.
+            #>
+    
+            '____' | Should -Be [regex]::Matches('Bears Beat Bongos', '(B.+){3}').Value
+    
+            $badgers = 'BadgerBadgerBadger'
+            $groupingPattern = '____'
+            [regex]::Matches($badgers, $groupingPattern).Value | Should -Be 'BadgerBadgerBadger'
+            $groupingPattern.Contains(')') | Should -BeTrue    # Use a group!
+    
+        }
+    }
+
+    Context '[ and ] - Square Brackets' {
+        It 'defines a set' {
+            <#
+                Square brackets denote a set, or array/collection, of symbols within a regular
+                expression. Imagine a pattern that might read "a or b or c or d". You may use square
+                brackets to create that regex. The set contained within the square brackets
+                represent one charactermwithin the string being searched.
+            #>
+    
+            [bool]$eolMatch = $____ # Should be either $true or $false
+            $eolMatch | Should -Be 'End of the line' -match '[efg]$'
+    
+            '____' -match '[abcdefg]$' | Should -BeTrue
+    
+            <#
+                Normally, the ^ character means "start of a line", but inside of square brackets, it
+                negates the set. Instead of "these symbols" it means "not these symbols"
+            #>
+            '____' -match '[^abcdefg]$' | Should -BeTrue
+    
+            <#
+                What do you think you might do if you want to match a literal ^ symbol within a set?
+                Use the ^ (caret) symbol in this one.
+            #>
+            $caretMatch = '[____]'
+            '^&*' -match $caretMatch | Should -BeTrue
+            $caretMatch.Contains('^') | Should -BeTrue      # Use the caret symbol!
+        }
+    }
+}
+
+Describe 'Meditative Examples' {
+    <#
+        Here are some challenges for you to put your new skills to work on.
+    #>
+    
+    Context 'Isolate a Username From a domain\username String Using Just One Regex Pattern' {
+        It 'is handy for isolating parts of strings' {
+            $usernamePattern = '____'
+            [regex]::Matches('cat\harley', $usernamePattern).Value | Should -Be 'harley'
+            [regex]::Matches('cat\kali', $usernamePattern).Value | Should -Be 'kali'
+            [regex]::Matches('human\thomas', $usernamePattern).Value | Should -Be 'thomas'
+        }
```

Comment:
Good place to use `-TestCases` to make this a little cleaner. Also spacing around the `It`s

Reply:
Are test cases used elsewhere in PSKoans?

Reply:
Hmm, not sure. If they are, it's fairly rare.

Not a bad place to do it though imo, since it doesn't really require the user to know precisely how to use them, the blank's just in the main `It` body.

Reply:
I'm just a little hesitant to diverge from how the rest of the koans look and feel. This syntax is less efficient, but makes it really clear to someone who doesn't know as much about what they're looking at to tell what's going on. This seems like a weird place to start using test cases.

Reply:
Agreed. Having slept on it, it's not a good way to go here. There's limited support for it in my koan-counting functions, so it's probably better to avoid it.

Thanks! 😊